src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -258,9 +258,9 @@ internal WatermarkOffsets GetWatermarkOffsets(string topic, int partition)
             return new WatermarkOffsets(low, high);
         }
 
-        internal void Subscribe(ICollection<string> topics)
+        internal void Subscribe(IEnumerable<string> topics)
         {
-            IntPtr list = LibRdKafka.topic_partition_list_new((IntPtr) topics.Count);
+            IntPtr list = LibRdKafka.topic_partition_list_new((IntPtr) topics.Count());
```

Comment:
I understand the reason for ICollection, but exposes the internal implementation of the method in a very minor way by showing that the collection requires to be counted. Having thought about it for a while, I'm not sure what is best..

I suppose that inside the method, to get the count you could do something like
`var partitionCollection = partitions as ICollection ?? partitions.ToArray();`

This would allow the method to support IEnumerable, but automatically handle deferred execution types. BUT, it seems like a lot of extra effort/maintenance just for the simple benefit of supporting IEnumerable.